internal/controller/isbservicerollout_controller.go

```diff
@@ -400,8 +413,19 @@ func (r *ISBServiceRolloutReconciler) isISBServiceUpdating(ctx context.Context,
 }
 
 func (r *ISBServiceRolloutReconciler) getPipelineList(ctx context.Context, rolloutNamespace string, rolloutName string) ([]*kubernetes.GenericObject, error) {
-	// here
-	return kubernetes.ListCR(ctx, r.restConfig, common.NumaflowAPIGroup, common.NumaflowAPIVersion, "pipelines", rolloutNamespace, fmt.Sprintf("%s=%s,%s", common.LabelKeyISBServiceNameForPipeline, rolloutName, common.LabelKeyPipelineRolloutForPipeline), "")
+	var pipelineList *unstructured.UnstructuredList
+	if err := r.client.List(ctx, pipelineList, client.InNamespace(rolloutNamespace), client.MatchingLabels{common.LabelKeyPipelineRolloutForPipeline: rolloutName}); err != nil {
+		return nil, err
+	}
+	objects := make([]*kubernetes.GenericObject, len(pipelineList.Items))
+	for i, unstruc := range pipelineList.Items {
+		obj, err := kubernetes.UnstructuredToObject(&unstruc)
+		if err != nil {
+			return nil, err
+		}
+		objects[i] = obj
+	}
+	return objects, nil
```

Comment:
So, I assume that since we aren't adding Unstructured type to the cache for Pipelines yet, then this is going to K8S API?

Reply:
Yes, For pipeline currently it's going to k8s API